### PR TITLE
parse nonisolated(unsafe) as decl contextual keyword in top-level global scope rather than as function invocation

### PR DESCRIPTION
**Pull request by sophiapoirier**
(temporary solution)

### CODE REVIEW

**Comment by sophiapoirier** [comment]
@swift-ci please test

**Comment by sophiapoirier** [comment]
@swift-ci please build toolchain

**Comment by sophiapoirier** [comment]
@swift-ci please test macOS

**Comment by sophiapoirier** [comment]
@swift-ci please build toolchain macOS

**Comment by sophiapoirier** [comment]
@swift-ci please build toolchain macOS

**Comment by sophiapoirier** [comment]
@swift-ci please build toolchain macOS

**Comment by sophiapoirier** [comment]
@swift-ci please build toolchain macOS

**Comment by sophiapoirier** [comment]
@swift-ci please build toolchain macOS

File: lib/Parse/ParseDecl.cpp [hunk 1]
```diff
@@ -5435,10 +5435,29 @@ static void skipAttribute(Parser &P) {
...
 bool Parser::isStartOfSwiftDecl(bool allowPoundIfAttributes,
                                 bool hadAttrsOrModifiers) {
+  const bool isTopLevelLibrary = (SF.Kind == SourceFileKind::Library) ||
+                                 (SF.Kind == SourceFileKind::Interface) ||
+                                 (SF.Kind == SourceFileKind::SIL);
   if (Tok.is(tok::at_sign) && peekToken().is(tok::kw_rethrows)) {
     // @rethrows does not follow the general rule of @<identifier> so
     // it is needed to short circuit this else there will be an infinite
     // loop on invalid attributes of just rethrows
+  } else if ((Tok.getKind() == tok::identifier) &&
```
Comment (hborla) [inline lib/Parse/ParseDecl.cpp:5445]: Should this logic be gated behind your experimental feature flag?

**Review by hborla — APPROVED** [review]

File: lib/Parse/ParseDecl.cpp (hunk 1 above)
Comment (sophiapoirier) [reply lib/Parse/ParseDecl.cpp:5445]: ah yes indeed...

**Comment by sophiapoirier** [comment]
@swift-ci please test

**Comment by sophiapoirier** [comment]
@swift-ci please test Windows

**Pull request merged**